01_lecture/homework/task2.js

```diff
@@ -0,0 +1,17 @@
+function wrapper(fn) {
+  const bufferOfCached = [];
+  return function () {
+    resultOfOperation = fn.apply(this, arguments);
+    let foundObj = bufferOfCached.find(
+      (x) =>
+        x.result === resultOfOperation &&
+        x.args.join("") === [...arguments].join("")
+    );
+    if (foundObj) {
+      return foundObj.result;
+    } else {
+      bufferOfCached.push({ args: [...arguments], result: resultOfOperation });
+      return resultOfOperation;
+    }
```

Comment:
You can avoid extra conditional branching

```suggestion
    if (foundObj) {
      return foundObj.result;
    }
    bufferOfCached.push({ args: [...arguments], result: resultOfOperation });
    return resultOfOperation;
   
```

Reply:
done